ops/ccs-ops-misc/synthetic-data/scripts/synthea-automation/README.md

```diff
@@ -0,0 +1,66 @@
+# Synthea Automation Scripts and Docker Image
+
+This directory contains several Synthea-related scripts pertaining to generating synthetic data
+using Synthea, as well as the definition for a Docker image (`bfd-mgmt-synthea-generation`) that
+includes said scripts.
+
+## Generating Synthea Data Locally Using the `bfd-mgmt-synthea-generation` Docker Image
+
+It is _preferred_ to generate Synthea data locally using the `bfd-mgmt-synthea-generation` Docker
+image, as it has the following advantages over using the `run_synthea_locally.sh` script:
+
+1. The Docker image comes pre-built with the necessary mapping files, the Synthea executable, and
+   all other dependencies
+2. Using the Docker image does not affect your local filesystem
+3. Using the Docker image does not require connecting to AWS or any external source; all data
+   generation is done locally to your machine
+
+Usage is simple:
+
+1. Ensure you are authenticated with AWS and are able to pull images from our ECR as well as from S3
+2. Pull the latest `end_state.properties` file from S3:
+
+   ```bash
+   aws s3 cp "s3://bfd-mgmt-synthea/end_state/end_state.properties" .
+   ```
+
+3. Pull the `bfd-mgmt-synthea-generation` image from ECR:
+
+   ```bash
+   AWS_REGION="us-east-1"
+   PRIVATE_REGISTRY_URI="$(aws ecr describe-registry --region "$AWS_REGION" | jq -r '.registryId').dkr.ecr.${AWS_REGION}.amazonaws.com"
+   IMAGE_NAME="${PRIVATE_REGISTRY_URI}/bfd-mgmt-synthea-generation"
+
+   docker pull "$IMAGE_NAME"
+   ```
+
+4. Run the image as a container, _bind-mounting_ the various directories (`output/`, `logs/`) as
+   well as the `end_state.properties` file downloaded in step (2):
+
+   ```bash
+   NUMBER_OF_BENES=100
+   NUMBER_OF_FUTURE_MONTHS=0
+   docker run -v "$(pwd)"end_state.properties:/usr/local/synthea/end_state.properties:ro \n
+    -v "$(pwd)"/logs:/usr/local/synthea/logs \ 
+    -v "$(pwd)"/output:/usr/local/synthea/output "$IMAGE_NAME:latest" -n "$NUMBER_OF_BENES" -f "$NUMBER_OF_FUTURE_MONTHS"
+   ```
+
+5. When finished, the generated output should be available in your current working directory in the
+   `output` sub-directory
+
+## Generating Synthea Data Using the `bfd-run-synthea-generation` Jenkins Pipeline
+
+This method is _preferred_ if you are looking to generate data that will be loaded into any of the
+BFD Server's environments. Using this will automatically upload the Synthea data into S3 for easy
+loading into the Pipeline.
```

Comment:
Given this is the more common case, maybe this should be first? The point of the jenkins job is to not need to run it locally :)

Reply:
I agree with this. I'll switch them around in the `README`.

Reply:
I've gone ahead and swapped their order in the `README`.